Add support for optional aws credentials for s3 for ingestion (#9375)
apache/druid: docs/ingestion/native-batch.md

|uris|JSON array of URIs where S3 objects to be ingested are located.|None|`uris` or `prefixes` or `objects` must be set|
|prefixes|JSON array of URI prefixes for the locations of S3 objects to be ingested.|None|`uris` or `prefixes` or `objects` must be set|
|objects|JSON array of S3 Objects to be ingested.|None|`uris` or `prefixes` or `objects` must be set|

> |properties|Properties Object for overriding default S3 configuration. See below for more information.|None|No (Default configurations will be use if not given)

Reviewer: ```suggestion
|properties|Properties Object for overriding the default S3 configuration. See below for more information.|None|No (defaults will be used if not given)
```

Author: Done

|bucket|Name of the S3 bucket|None|yes|
|path|The path where data is located.|None|yes|

Properties Object:

|property|description|default|required?|
|--------|-----------|-------|---------|

> |accessKeyId|S3 access key for this S3 InputSource|None|yes if secretAccessKey is given|

Reviewer 2: I think these can just be shortened to `S3 access key` and `S3 secret key`

Author: What do you mean? The description or the property itself?

Author: Oh i see. Wanted to emphasize that these configs will only be applied to this input source (not the deep storage, etc.)

|secretAccessKey|S3 secret key for this S3 InputSource|None|yes if accessKeyId is given|

> **Note :** *If any property is not given in the Properties Object, then the current default server value will be use.*

Reviewer: ```suggestion
Default values are used for any property not overridden in the Properties Object.
```
Caveat to my edit suggestion: it says "any property" but really it's just S3 access key that are supported, no? To avoid the premature generalization, could say something like: 

"If you don't provide an access key, the default S3 credentials provider chain is used." (With a link to the relevant section in s3.md).

Author: That is the idea for the future when we support more than just the access key and secret key. Will changed to "If you don't provide an access key, the default S3 credentials provider chain is used." for now and we can change it back in the future.

Author: done
